### PR TITLE
feat: add create group page

### PR DESCRIPTION
**Pull request by karantakalkar**
## Description
Added routing to 'Create Group' button in Group's List View, designed the UI for Create Group Form
and connected it to the API.

## Related issues and discussion
#212 

## Screenshots, if any
**UI**

Screenshot from 2020-02-08 18-49-31

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by karantakalkar** [comment]
Data Tables are not functional in demo.mifos.io beacuse the API response (post create group button-presw) has no field for data-tables,
Could not find reference to code out data-tables feature.

File: src/app/groups/create-group/create-group.component.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      this.groupForm.patchValue({
+        submittedOnDate: this.datePipe.transform(submittedOnDate, dateFormat),
+        activationDate: this.datePipe.transform(activationDate, dateFormat)
+      });
+      const group = this.groupForm.value;
+      group.locale = 'en';
+      group.dateFormat = dateFormat;
+      group.clientMembers = [];
+      this.clientMembers.forEach(client => group.clientMembers.push(client.id));
+      console.log(group);
```
Comment (karantakalkar) [inline src/app/groups/create-group/create-group.component.ts]: need to remove this

File: src/app/groups/create-group/create-group.component.html [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            <mat-label>Submitted on</mat-label>
+            <input matInput [min]="minDate" [max]="maxDate" [matDatepicker]="submittedOnDatePicker" required formControlName="submittedOnDate">
+            <mat-datepicker-toggle matSuffix [for]="submittedOnDatePicker"></mat-datepicker-toggle>
+            <mat-datepicker #submittedOnDatePicker></mat-datepicker>
+            <mat-error *ngIf="groupForm.controls.submittedOnDate.hasError('required')">
+              Submission Date is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
```
Comment (karantakalkar) [inline src/app/groups/create-group/create-group.component.html]: need to replace this by mat autocomplete

File: src/app/groups/create-group/create-group.component.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    });
+    this.buildDependencies();
+  }
+
+  buildDependencies() {
+    /**
+     * Sets the staff and clients data each time the user selects a new office.
+     */
+    this.groupForm.get('officeId').valueChanges.subscribe((option: any) => {
+
```
Comment (karantakalkar) [inline src/app/groups/create-group/create-group.component.ts]: remove line

File: src/app/groups/groups.service.ts [hunk 4]
```diff
@@ -14,4 +14,15 @@ export class GroupsService {
         return  this.http.get('/groups');
     }
 
+    getStaff(id: number) {
+        const httpParams = new HttpParams()
+            .set('officeId', id.toString())
+            .set('staffInSelectedOfficeOnly', 'true');
+        return this.http.get('/centers/template', { params: httpParams });
```
Comment (karantakalkar) [inline src/app/groups/groups.service.ts]: /groups/template

File: src/app/groups/create-group/create-group.component.ts [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+   * Retrieves the offices data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   * @param {ClientsService} clientsService CentersService.
+   * @param {GroupsService} groupService GroupsService.
+   * @param {DatePipe} datePipe Date Pipe to format date.
+   */
+  constructor(private formBuilder: FormBuilder,
+    private route: ActivatedRoute,
```
Comment (karantakalkar) [inline src/app/groups/create-group/create-group.component.ts]: indent here

File: src/app/groups/create-group/create-group.component.html [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+              <div class="mat-cell">Office</div>
+              <div class="mat-cell">{{clientChoice.value.officeName}}</div>
+            </div>
+          </div>
+
+          <mat-nav-list *ngIf="clientMembers.length">
+            <h3 matSubheader>Selected Clients</h3>
+            <div mat-list-item *ngFor="let client of clientMembers index as i">
+              <button mat-icon-button (click)="removeClient(i)">
+                <fa-icon icon="trash"></fa-icon>
```
Comment (muskankhedia) [inline src/app/groups/create-group/create-group.component.html:112]: Please shift the icon to the right end, that will provide a better view

File: src/app/groups/create-group/create-group.component.html [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            <mat-error *ngIf="groupForm.controls.officeId.hasError('required')">
+              Office is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field>
+            <mat-label>Staff</mat-label>
+            <mat-select formControlName="staffId">
+              <mat-option *ngFor="let staff of staffData" [value]="staff.id">
+              {{ staff.displayName }}
```
Comment (muskankhedia) [inline src/app/groups/create-group/create-group.component.html:38]: You can deactivate the staff field if staffData.length === 0. That will provide a better user experience.

**Comment by karantakalkar** [comment]
@muskankhedia there are data tables to be implemented in this component as per communtiy app code, however they do not work in demo.mifos.io/staging.mifos.io, It functions similar to these servers though.

**Comment by muskankhedia** [comment]
Okay, that's fine. You can raise a JIRA ticket for the same if there is an error from the backend. Till then you can resolve the front-end issues.

**Pull request merged**